test/test_basic_submission.py

```diff
@@ -109,36 +110,47 @@ def test_dockstore_import_in_terra(self):
+        try:
+            response = run_workflow()
             status = response['status']
-            if response['workflows'][0]['status'] == "Failed":
-                raise RuntimeError(f'The md5sum workflow did not succeed:\n{json.dumps(response, indent=4)}')
-
-            print(f"md5sum workflow state is: {response['workflows'][0]['status']}.  Checking again in 20 seconds.")
-            if timeout < 0:
-                print(json.dumps(response, indent=4))
-                raise RuntimeError('The md5sum workflow run timed out.  '
-                                   f'Expected 4 minutes, but took longer than '
-                                   f'{float(twenty_minutes - timeout) / 60.0} minutes.')
-
-        with self.subTest('Dockstore Workflow Run Completed Successfully'):
-            if response['workflows'][0]['status'] != "Succeeded":
-                raise RuntimeError(f'The md5sum workflow did not succeed:\n{json.dumps(response, indent=4)}')
+            with self.subTest('Dockstore Workflow Run Submitted'):
+                self.assertEqual(status, 'Submitted')
+            with self.subTest('Dockstore Workflow Run Responds with DRS.'):
+                self.assertTrue(response['workflows'][0]['inputResolutions'][0]['value'].startswith('drs://'))
+
+            submission_id = response['submissionId']
+
+            # md5sum should run for about 4 minutes, but may take far longer(?); give a generous timeout
+            # also configurable manually via MD5SUM_TEST_TIMEOUT if held in a pending state
+            start = time.time()
+            deadline = start + int(os.environ.get('MD5SUM_TEST_TIMEOUT', 60 * 60))
+            while True:
+                response = check_workflow_status(submission_id=submission_id)
+                status = response['status']
+                if status == 'Done':
+                    break
+                elif response['workflows'][0]['status'] == "Failed":
+                    raise RuntimeError(f'The md5sum workflow did not succeed:\n{json.dumps(response, indent=4)}')
+                else:
+                    now = time.time()
+                    if now < deadline:
+                        print(f"md5sum workflow state is: {response['workflows'][0]['status']}. "
+                              f"Checking again in 20 seconds.")
+                        time.sleep(20)
+                    else:
+                        print(json.dumps(response, indent=4))
+                        raise RuntimeError('The md5sum workflow run timed out.  '
+                                           f'Expected 4 minutes, but took longer than '
+                                           f'{float(start - now) / 60.0} minutes.')
+
+            with self.subTest('Dockstore Workflow Run Completed Successfully'):
+                if response['workflows'][0]['status'] != "Succeeded":
+                    raise RuntimeError(f'The md5sum workflow did not succeed:\n{json.dumps(response, indent=4)}')
+        finally:
```

Comment:
We should only post the time if we get a successful workflow submission.  Otherwise it's going to be posting very quick latency times whenever the test fails, which it shouldn't do.

Reply:
Hmm interest point. Quick failures like that could be confusing in the graph. If a workflow queues for 15min then fails, wouldn't we want to capture that information though?

Reply:
Yeah, we should capture failures and successes, but only if the initial workflow request succeeds.  We want to measure the time it's queued and running.  If it's never queued, we should discard that though, since the workflow never ran and that's not a measure of latency.